### DISCUSSION TITLE
read options from file

### DISCUSSION THREAD

**Discussion by ghost** [mitmproxy/mitmproxy#6344] [General]
sorry for the dumb question. I have read the documentation and looked around the repo as well, but I cant seem to answer this simple question:

I want MITM Proxy to read options from a file, instead of having to set each one. so I want to set `block_list` and `modify_headers` and similar, but again from a file that MITM Proxy will read, instead of manually having to set them in the program itself. is this possible, if so how? and where is this documented?

**Comment by mhils** [comment]
Check the first paragraph at
https://docs.mitmproxy.org/stable/concepts-options/. :-)

**Reply by ghost** [reply]
sorry I am an idiot. I did read that, but I am still not understanding:

1. option can ONLY be read from that file? can I specify another?
2. if I can specify another, how do I do that?
3. what is the syntax of that file, I didn't see an example

**Reply by mhils** [reply]
1. You can specify --confdir to load config.yaml from a different directory. But that changes the entire configuration directory.
2. See above.
3. It's YAML - but I agree we should add a small example! :)
```yaml
anticache: true
```

**Reply by ghost** [reply]
I dont think `--confdir` is a good solution, since changing also causes MITM Proxy to generate certificates in whatever folder you change it to. if I want to keep separate files for different apps, it looks like using an addon might be a better option. for example, here is something similar to `block_list` and `modify_headers`:

~~~py
from mitmproxy import http

def request(f: http.HTTPFlow) -> None:
   if f.request.path.startswith('/play/log'):
      f.kill()
   if f.request.path.startswith('/fdfe/acquire'):
      f.request.headers.pop('Connection', None)
~~~
